app/Models/Application.php

```diff
@@ -272,6 +297,14 @@ public function setStatusAttribute(ApplicationStatus|string $status)
         if ($status === ApplicationStatus::CheckedIn) {
```

Comment:
Especially the three statuses `TableAssigned` <-> `CheckedIn` <-> `CheckedOut` should never be navigated outside this order in everyday operation. We should restrict any status jumps outside of that order in the admin UI in the future instead to avoid unexpected status changes.